crypto/s390xcap.c

```diff
@@ -508,6 +529,101 @@ static int parse_env(struct OPENSSL_s390xcap_st *cap)
                        | S390X_CAPBIT(S390X_AES_192)
                        | S390X_CAPBIT(S390X_AES_256),
                        0ULL},
+        /*.pcc    = */{S390X_CAPBIT(S390X_QUERY),
```

Comment:
Just out of interest - why do you set the QUERY here?

Reply:
These structures map the most recent architecture to real-worl processors that is, every structure indicates which (openssl-relevant) instructions and function codes are available with the corresponding processor.

> Would it make much difference really if this was just NULL?

It would mean pcc instruction's query function code is not available on z15, which is false.
However, it wouldnt make a difference for openssl's capability detection: The structures are only used with OPENSSL_s390xcap environment variable, providing the possibility to mask/turn-off features which were detected on the machine. See OPENSSL_s390xcap(3). This is useful eg for regression testing, see https://github.com/openssl/openssl/issues/9262.